File: Codes/2789-largest-element-in-an-array-after-merge-operations.cpp
/*
 * @Author: LetMeFly
 * @Date: 2024-03-14 08:51:31
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-03-14 08:54:31
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

typedef long long ll;
class Solution {
public:
    ll maxArrayValue(vector<int>& nums) {
        ll ans = nums.back();
        for (int i = nums.size() - 2; i >= 0; i--) {
            if (nums[i] <= ans) {
                ans += nums[i];
            }
            else {
                ans = nums[i];
            }
        }
        return ans;
    }
};

File: Codes/2789-largest-element-in-an-array-after-merge-operations.py
'''
Author: LetMeFly
Date: 2024-03-14 08:55:17
LastEditors: LetMeFly
LastEditTime: 2024-03-14 08:56:09
'''
from typing import List

class Solution:
    def maxArrayValue(self, nums: List[int]) -> int:
        ans = nums[-1]
        for i in range(len(nums) - 2, -1, -1):
            if nums[i] <= ans:
                ans += nums[i]
            else:
                ans = nums[i]
        return ans